src/SignalR/server/Core/src/HubOptionsExtensions.cs

```diff
@@ -0,0 +1,31 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System.Collections.Generic;
+using Microsoft.AspNetCore.SignalR.Internal;
+
+namespace Microsoft.AspNetCore.SignalR
+{
+    public static class HubOptionsExtensions
+    {
+        public static void AddFilter(this HubOptions options, IHubFilter hubFilter)
+        {
+            if (options.HubFilters == null)
+            {
+                options.HubFilters = new List<IHubFilter>();
+            }
+
+            options.HubFilters.Add(hubFilter);
+        }
+
+        public static void AddFilter<T>(this HubOptions options) where T : IHubFilter
```

Comment:
1. What is the lifetime of instances of T? Created per-invocation?
2. Have you considered allowing arguments to be specified? That's something supported by gRPC interceptors https://github.com/grpc/grpc-dotnet/blob/master/src/Grpc.AspNetCore.Server/InterceptorCollection.cs
3. Non-generic `AddFilter(Type filterType)` overload?

Reply:
> What is the lifetime of instances of T? Created per-invocation?

It's up to the implementation. Either singleton or per invocation. We do the service provider then object factory trick.

> Have you considered allowing arguments to be specified? That's something supported by gRPC interceptors https://github.com/grpc/grpc-dotnet/blob/master/src/Grpc.AspNetCore.Server/InterceptorCollection.cs

That seems reasonable or we could support a factory.

> Non-generic AddFilter(Type filterType) overload?

👍

Reply:
```suggestion
        public static void AddFilter<THubFilter>(this HubOptions options) where THubFilter : IHubFilter
```